### PR TITLE
fix: Fix actor_id constraint migration to account for duplicates

### PR DESCRIPTION
**Pull request by markstory**
When reworking the foreign keys on the actor table we missed a race condition that resulted in 96 users getting an duplicate actor generated. Each actor that is generated would update `user.actor_id`. The value in `user.actor_id` is used to set notification settings and attach actors to issues.

The additional query performs the following steps:

1. Find users that have more than one actor.
2. Combine actor + user to find the actors that aren't the authoritative ones. These records can be identified by the left join not matching.
3. Null out the user_id for all results from the join failure in 2. I was concerned about integrity errors from other usage, and nulling out actor.user_id is how the application code works too.

We aren't creating any *new* duplicate actors so these changes should resolve the integrity errors blocking constraint creation.

### CODE REVIEW

**Comment by vbro** [comment]
LGTM. Confirmed that there are 192 duplicates, which come in 96 pairs of `user_id`s. For each pair, one has a corresponding User (via auth_user.actor_id) and the other doesn't.

File: src/sentry/migrations/0418_add_actor_constraints.py [hunk 1]
```diff
@@ -24,6 +24,24 @@ class Migration(CheckedMigration):
 
     operations = (
         [
+            migrations.RunSQL(
+                sql="""
+WITH duplicate_users AS (
+  SELECT user_id FROM sentry_actor WHERE user_id IS NOT NULL GROUP BY user_id HAVING count(*) > 1
+),
+cleanup_actors AS (
+    SELECT * FROM sentry_actor AS actor
```
Comment (vbro) [inline src/sentry/migrations/0418_add_actor_constraints.py:33]:
```suggestion
    SELECT actor.id FROM sentry_actor AS actor
```

Comment (vbro) [reply src/sentry/migrations/0418_add_actor_constraints.py:33]: This avoid the `AmbiguousColumn` error.

**Review by vbro — APPROVED** [review]

**Review by wedamija — COMMENTED** [review]
This looks ok to me, is ops comfortable running bare update scripts on postgres? Usually they only do that for indexes/constraints etc

File: src/sentry/migrations/0418_add_actor_constraints.py (hunk 1 above)
Comment (markstory) [reply src/sentry/migrations/0418_add_actor_constraints.py:33]: Thanks!

**Comment by markstory** [comment]
> This looks ok to me, is ops comfortable running bare update scripts on postgres? Usually they only do that for indexes/constraints etc

Ben is on-board with running this so we can unblock this migration and its dependencies.

**Review by wedamija — APPROVED** [review]

**Pull request merged**